azurerm/internal/services/containers/container_group_resource.go

```diff
@@ -418,6 +418,37 @@ func resourceArmContainerGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+
+			"dns_config": {
+				Optional: true,
+				MaxItems: 1,
+				ForceNew: true,
+				Type:     schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"nameservers": {
+							Type:     schema.TypeList,
```

Comment:
Can we change the type into `schema.TypeSet` as azure API doesn't always guarantee the order to be same as in request and in response when later get.